easybuild/easyconfigs/h/HDF5/HDF5-1.10.1-PGI-18.4-GCC-6.4.0-2.28.eb

```diff
@@ -0,0 +1,24 @@
+name = 'HDF5'
+version = '1.10.1'
+
+homepage = 'https://support.hdfgroup.org/HDF5/'
+description = """HDF5 is a data model, library, and file format for storing and managing data.
+ It supports an unlimited variety of datatypes, and is designed for flexible
+ and efficient I/O and for high volume and complex data."""
+
+toolchain = {'name': 'PGI', 'version': '18.4-GCC-6.4.0-2.28'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-%(version_major_minor)s/hdf5-%(version)s/src']
+sources = [SOURCELOWER_TAR_GZ]
+
+checksums = ['048a9d149fb99aaa1680a712963f5a78e9c43b588d0e79d55e06760ec377c172']
+
+buildopts = "CXXFLAGS='--noswitcherror'"
```

Comment:
@SimonPinches Can you clarify why this is required? Maybe we should include a comment in these easyconfig to explain it? Never seen this before (but my experience with PGI is limited).

Reply:
Actually, this is even causing problems for me...

```
pgc++-Error-Unknown switch: --noswitcherror
```

Reply:
It's definitely a clean way to fix the problem, but I think it's worth clarifying it by including a comnent in the easyconfig itself? You're hard overriding the `$CXXFLAGS` set by EasyBuild now, which is probably not the intention, since you only want to *add* a flag?

```python
# add -noswitcherror to avoid that libtool adds -pthread (which is not known by PGI compiler)
buildopts = 'CXXFLAGS="$CXXFLAGS -noswitcherror"'
```

Note: the switching of quotes is required to ensure that `$CXXFLAGS` gets expanded when the build command is run by EasyBuild...